pyanaconda/core/service.py

```diff
@@ -0,0 +1,139 @@
+#
+# Copyright (C) 2021  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from pyanaconda.core.util import execWithRedirect, execWithCapture
+from pyanaconda.anaconda_loggers import get_module_logger
+
+log = get_module_logger(__name__)
+
+__all__ = ["start_service", "stop_service", "restart_service", "is_service_running",
+           "is_service_installed", "enable_service", "disable_service"]
+
+
+def _run_systemctl(command, service, root):
+    """Runs 'systemctl command service'
+
+    :param str command: command to run on the service
+    :param str service: name of the service to work on
+    :param str root: root to run the command in
+    :return: exit status of the systemctl run
+    """
+
+    args = [command, service]
+    if root != "/":
+        args += ["--root", root]
+
+    ret = execWithRedirect("systemctl", args)
+
+    return ret
+
+
+def start_service(service):
+    """Start a systemd service in the installation environment
+
+    Runs 'systemctl start service'.
+
+    :param str service: name of the service to start
+    :return: exit status of the systemctl run
+    """
+    return _run_systemctl("start", service, "/")
+
+
+def stop_service(service):
+    """Stop a systemd service in the installation environment
+
+    Runs 'systemctl stop service'.
+
+    :param str service: name of the service to stop
+    :return: exit status of the systemctl run
+    """
+    return _run_systemctl("stop", service, "/")
+
+
+def restart_service(service):
+    """Restart a systemd service in the installation environment
+
+    Runs 'systemctl restart service'.
+
+    :param str service: name of the service to restart
+    :return: exit status of the systemctl run
+    """
+    return _run_systemctl("restart", service, "/")
+
+
+def is_service_running(service):
+    """Is a systemd service running in the installation environment?
+
+    Runs 'systemctl status service'.
+
+    :param str service: name of the service to check
+    :return: was the service found
+    """
+    ret = _run_systemctl("status", service, "/")
+
+    return ret == 0
+
+
+def is_service_installed(service, root):
```

Comment:
I was considering to make the `root` argument mandatory, but when you combine the functions with and without the `root` argument, it is just confusing and looks like a bug in the code. 

```
if is_service_installed(RHSM_SERVICE_NAME, "/"):
  start_service(RHSM_SERVICE_NAME)
```

I still think that using `root="/"` by default will more consistent with our libraries.